src/main/java/com/exasol/bucketfs/ReadEnabledBucket.java

```diff
@@ -134,16 +133,13 @@ public void downloadFile(final String pathInBucket, final Path localPath)
 
     // [impl->dsn~downloading-a-file-from-a-bucket-as-string~1]
     @Override
-    public String downloadFileAsString(final String pathInBucket)
-            throws InterruptedException, BucketAccessException {
+    public String downloadFileAsString(final String pathInBucket) throws InterruptedException, BucketAccessException {
         final URI uri = createPublicReadURI(pathInBucket);
-        LOGGER.fine(() -> "Downloading  file from bucket \"" + this.bucketFsName + "/" + this.bucketName + "\": \""
-                + uri + "\"");
+        LOGGER.fine(() -> "Downloading  file from bucket \"" + this + "\" at \"" + uri + "\"");
         try {
             return httpGet(uri);
         } catch (final IOException exception) {
-            throw new BucketAccessException(
-                    "Unable to download file from BucketFS as string.", uri, exception);
+            throw new BucketAccessException("Unable to download file from BucketFS as string.", uri, exception);
```

Comment:
I think you missed ExaError builder here?

Reply:
1. I forgot the whole uploading part is in a different class.
2. I am changing the exceptions in a separate PR.